app_dart/lib/src/request_handlers/github/webhook_subscription.dart

```diff
@@ -423,13 +423,14 @@ class GithubWebhookSubscription extends SubscriptionHandler {
           // for the purposes of testing a change that otherwise needs tests,
           // but since they are the driver for tests they don't need test
           // coverage.
-          !filename.endsWith('tool/run_tests.dart') &&
           !filename.endsWith('run_tests.sh')) {
         needsTests = !_allChangesAreCodeComments(file);
       }
       // See https://github.com/flutter/flutter/wiki/Plugin-Tests for discussion
       // of various plugin test types and locations.
       if (filename.endsWith('_test.dart') ||
+          // Test files in custom package-specific test folders.
+          filename.contains(RegExp('packages/packages/[^/]+/tool/')) ||
```

Comment:
It's not the case that everything in the tool directory is a test, so this is too broad. `tool` is just the convention for non-public scripts: https://dart.dev/tools/pub/package-layout#public-tools

E.g., a code generation script might live in `tool`, but not be a test.

Reply:
What about we match any file that has word `test` anywhere in this folder? something like this
'packages/packages/[^/]+/tool/.*test.*'

Or do you prefer more package specific exception here? like
`packages/packages/go_router/tool/test_fixes`

Reply:
> What about we match any file that has word `test` anywhere in this folder? something like this
> 'packages/packages/[^/]+/tool/.+test.+'

There are currently only three examples of that pattern, at at least one is a false positive (`pigeon/tool/test.dart`, which is just a script for running manual tests locally, so provides no automated test coverage).

The missing-test bot should err on the side of false positives rather than false negatives.

Reply:
sounds good I will make this more specific to package instead. If we have more and more use cases, we can come up with a more systematic solution later.